### PR TITLE
fix: utils/utils.go: fix create boilerplate directoy to current working directory on Windows and  refs other XDG_DATA_HOME and LOCALAPPDATA

### PR DESCRIPTION
**Pull request by atsushifx**
On Windows, create boiler plate directory to current working directory.
  move this directory to LOCALAPPDATA directory
ref
add: utils/utils.go: boiler plate directory refs "XDG_DATA_HOME"
  create boiler directory under the XDG_DATA_HOME directory,
  if your os set set XDG Base directory.

add: .editorconfig:
  editor tab setting file for go source edit.

### CODE REVIEW

File: utils/utils.go [hunk 1]
```diff
@@ -6,18 +6,35 @@ import (
...
+
 	override := os.Getenv("GIBO_BOILERPLATES")
 	if len(override) > 0 {
 		return override
 	}
-	return filepath.Join(os.Getenv("HOME"), ".gitignore-boilerplates")
+
+	override = os.Getenv("XDG_DATA_HOME")
+	if len(override) > 0 {
+		return filepath.Join(override, "gibo", gitignoreDir)
```
Comment (hyperupcall) [inline utils/utils.go:26]: Sorry for the drive-by review, but I'd like to point out that directories in `$XDG_DATA_HOME`, `$XDG_CONFIG_HOME`, etc. should not be hidden. So it should be something like `path.join(XDG_DATA_HOME, 'gibo', 'gitignore-boilerplates')`

**Comment by simonwhitaker** [comment]
Hi Atsushi! Thank you for contributing to gibo. And special thanks for contributing to gibo on Windows.

Can you please remove the editorconfig file from this PR and raise that as a separate PR? I'd like to merge your home dir fix, but I'd also like to think about the editorconfig change a bit more before merging that one.

**Comment by atsushifx** [comment]
I understood. I make new commit "[remove editor config](https://github.com/simonwhitaker/gibo/pull/62/commits/a193e3bbe885679acb5469ca8a54ed8b89bc216b)", then this PR have only utils/utils.go file.
please merge this if you have no probllem.

**Comment by simonwhitaker** [comment]
Hey @atsushifx... I thought you'd like to know, I was inspired by your PR to also store the boilerplates in the right place on MacOS too (i.e. `~/Library/Caches`), and that in turn lead me to discover go's `os.UserCacheDir()`. I've incorporated that in https://github.com/simonwhitaker/gibo/commit/bd62924a589bd1720b8e264c1406f0316407e68e.

**Comment by atsushifx** [comment]
I read this commit, and I want to make 2 sugesstions.

1. commit bd6924 make boilerplates cache on gibo directory directly. add boilerplates directory to cache , if another data cache by future.

2. I use XDG Base Directory Environment on Windows.  But os.os.UserCacheDir() function don't return XDG_CACHE_HOME instead of LOCALAPPDATA, please get $XDG_CACHE_HOME directory on windows , if you can change this.
